docs: pact nirvana refresh for branchs/envs (#275)
pact-foundation/docs.pact.io: website/docs/pact_nirvana/step_3.md

## Tasks

1. Write a Pact test in your consumer project \(consult [the documentation](/implementation_guides/cli) for your chosen language\) and generate the pact.

> 2. Ship the pact file to the provider team via some manual mechanism (we will introduce the pact-broker in a later step)

Reviewer: ```suggestion
2. Ship the pact file to the provider team via some manual mechanism (we will introduce the concept of a pact broker in a later step)
```

If you are on the provider team, you don't own this test, although you can work with your consumer team to help them 
write it.  But ultimately the consumer owns these tests and drives the understanding of the contract.

Note that if at all possible this first test should not introduce the complexity of provider states. You can add that as a later step.

:::info
You can see an example and overview of a consumer test [here](https://docs.pact.io/5-minute-getting-started-guide#testing-the-order-web-consumer-project)
:::

### Manually run the provider verification test

When you run the consumer test and it runs successfully, it is going to generate a pact file (for more information you 
can read [this overview](https://docs.pact.io/getting_started/how_pact_works#consumer-testing)). This file records a series of interactions where the consumer sends a request and the producer returns an expected response.

Now you need to see if the provider behaves as the consumer expects. Ultimately this will happen automatically as 
part of running your provider test suite both locally and in CI/CD. 

There are two types of pact verification that will be performed

> 1. Pact verification by URL or filepath (This will be designed to be triggered by a webhook, when a consumer contract changes)

Reviewer: Shall we keep the use of "file path" consistent?

```suggestion
1. Pact verification by URL or file path (This will be designed to be triggered by a webhook, when a consumer contract changes)
```

2. Pact verification by consumer version selectors (This will be designed to run, whenever the provider codebase changes and runs in CI)

You can make sure it works correctly before setting up all that automation.  

> This step will perform verification type 1, allowing providers to verify pacts from a local file-path, or remote URL. This is useful for

Reviewer: ```suggestion
This step will perform verification type 1, allowing providers to verify pacts from a local file path, or remote URL. This is useful for
```
